src/main/jbake/content/docs/components/metrics/index.adoc

```diff
@@ -1258,7 +1258,380 @@ A reset event occurs some time between 90000 and 210000; consequently, we will g
 Note that we exclude the rate data point between 90000 and 210000 timestamps.
 
 === Downsampling
-*TODO*
+Downsampling is a query technique for reducing the number of data points that are sent back to the client. Why is this
+done? When a request is made to render a graph, the client specifies a date range. The number of data points that fall
+within that range can and will vary. We want to avoid sending back too many data points because an excessive number of
+data points does little to improve the visualization, slows down the rendering, and makes the UI less response which in
+turn makes the user experience worse overall. Downsampling is a way to return a predictable or fixed number of data
+points which facilitates better graphs and a better overall user experience.
+
+Hawkular Metrics provides several `/stats` endpoints that use downsampling.
+
+==== Buckets
+Data points are first grouped into buckets. A bucket can have zero or more data points, and a data point will be in at
+most one bucket. Aggregation functions are then applied to the data points in each bucket to produce a single,
+_bucketed_ data point.
+
+Let's look at a simple example to illustrate how data points are grouped.
+
+[grid="all"]
+.Data points
+|===
+|Data point |Timestamp
+|P~1~ |15:00
+|P~2~ |15:10
+|P~3~ |15:20
+|P~4~ |15:30
+|P~5~ |15:40
+|P~6~ |15:50
+|===
+
+We have six data points. The values are irrelevant for the example. We query with a date range of 15:00 to 16:00.
+We use for buckets to end up with:
+
+[grid="all"]
+.Buckets
+|===
+|Bucket |Data points
+|15:00 - 15:15 |P~1~, P~2~
+|15:15 - 15:30 |P~3~
+|15:30 - 15:45 |P~4~, P~5~
+|15:45 - 16:00 |P~6~
+|===
+
+The first thing to note is that a bucket expressed as a date range or duration in which the start time is inclusive and
+the end time is exclusive. If a data point's timestamp falls within that range, then the data point is grouped into
+that bucket. Different aggregation functions are applied depending on the metric type.
+
+===== Bucket Query Parameters
+There are two query parameters that are available with all `/stats` endpoints - `buckets` and `bucketDuration`. One and
+only one of them can be specified in a request. For the preceding example, we could end up with four buckets using
+either one these parameters.
+
+`buckets` specifies the exact number of buckets to use. For the preceding example, `buckets=4` will divide the time
+range into four buckets. A higher value increases the number of buckets which in turn reduces the number of data points
+per bucket.
+
+`bucketDuration` is a duration in milliseconds that specifies how wide or large each bucket is. For the preceding
+example, `bucketDuration=900000` specifies that each bucket is 15 minutes wide, which again yields four buckets. A
```

Comment:
I thought bucketDuration had to be followed by the unit. So here: '15mn'

Has something changed in this area?